FGEGraphics/UISystem/UIClickableElement.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FGEGraphics.UISystem;
+
+namespace FGEGraphics.UISystem;
+
+/// <summary>Represents a clickable UI element on the screen.</summary>
+public abstract class UIClickableElement : UIElement
+{
+    /// <summary>Ran when this element is clicked.</summary>
+    public event EventHandler Clicked;
+
+    /// <summary>Whether the mouse is hovering over this element.</summary>
+    public bool Hovered = false;
+
+    /// <summary>Whether this element is being clicked.</summary>
+    public bool Pressed = false;
+
+    // TODO: Maybe factor out into "UIInteractableElement" or something?
+    /// <summary>Whether this element can be interacted with. Don't set directly, use <see cref="Enabled"/>.</summary>
+    public bool Internal_Enabled = true;
```

Comment:
See Internal-struct recommended pattern in the Frenetic coding guidelines: https://github.com/FreneticLLC/FreneticUtilities/blob/master/CodingGuidelines.md